packages/react-select/src/internal/ScrollManager.js

```diff
@@ -1,30 +1,65 @@
 // @flow
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React, { PureComponent, type Element } from 'react';
+import ScrollLock from './ScrollLock/index';
 
-import React, { Component, type Element } from 'react';
+type RefCallback<T> = (T | null) => void;
 
-import NodeResolver from './NodeResolver';
-
-export type CaptorProps = {
-  children: Element<*>,
+type Props = {
+  children: (RefCallback<HTMLElement>) => Element<*>,
+  lockEnabled: boolean,
+  captureEnabled: boolean,
   onBottomArrive?: (event: SyntheticEvent<HTMLElement>) => void,
   onBottomLeave?: (event: SyntheticEvent<HTMLElement>) => void,
   onTopArrive?: (event: SyntheticEvent<HTMLElement>) => void,
   onTopLeave?: (event: SyntheticEvent<HTMLElement>) => void,
+}
+
+type State = {
+  enableLock: boolean
+}
+
+const defaultProps = {
+  captureEnabled: true
 };
 
-class ScrollCaptor extends Component<CaptorProps> {
+export default class ScrollManager extends PureComponent<Props,State> {
+  static defaultProps = defaultProps;
+
   isBottom: boolean = false;
   isTop: boolean = false;
-  scrollTarget: HTMLElement;
   touchStart: number;
 
+  state = {
+    enableLock: false
+  }
+
+  targetRef = React.createRef<HTMLElement>();
+
+  blurSelectInput = () => document.activeElement && document.activeElement.blur();
+
   componentDidMount() {
-    this.startListening(this.scrollTarget);
+    if (this.props.captureEnabled) {
+      this.startListening(this.targetRef.current);
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.captureEnabled !== this.props.captureEnabled) {
+      if (this.props.captureEnabled) {
+        this.startListening(this.targetRef.current);
+      } else {
+        this.stopListening(this.targetRef.current);
+      }
+    }
   }
```

Comment:
Will this work if the DOM element changes?

Reply:
The problem has not occurred so far with `NodeResolver` and I think it should not be considered now. If it ever comes up, then we should come up with a solution.